Segfault when creating many pods simultaneously

**Describe the bug**
When a lot's of pods are being created simultaneously, vHive segfaults in `createUserContainer`.

**To Reproduce**
1. Setup a vHive cluster as specified in the quickstart guide (one master and one worker should be sufficient)
2. Modify `examples/deployer/functions.json` such that only helloworld will be deployed
3. Deploy helloworld by calling `go run examples/deployer/client.go`
4. Call `watch kubectl get pods` and wait until all pods of the helloworld deployment get deleted
5. Call `go run examples/invoker/client.go -rps 100`

**Logs**
```
WARN[2021-04-21T10:07:55.170793024-04:00] Using google dns 8.8.8.8
ERRO[2021-04-21T10:07:55.315457829-04:00] stock containerd failed to start UC           error="rpc error: code = Unknown desc = failed to create container io: failed to open /run/containerd/io.containerd.grpc.v1.cri/cont
ainers/ae4016be685c7b69fdab28942aeba1df87602b2115c6e5163a933556097af716/io/084543282/ae4016be685c7b69fdab28942aeba1df87602b2115c6e5163a933556097af716-stdout with O_PATH: open /run/containerd/io.containerd.grpc.v1.cri/con
tainers/ae4016be685c7b69fdab28942aeba1df87602b2115c6e5163a933556097af716/io/084543282/ae4016be685c7b69fdab28942aeba1df87602b2115c6e5163a933556097af716-stdout: too many open files"
ERRO[2021-04-21T10:07:55.376246971-04:00] stock containerd failed to start UC           error="rpc error: code = Unknown desc = failed to create container io: failed to open /run/containerd/io.containerd.grpc.v1.cri/cont
ainers/a16675f95a3f3c6b61307357be3c32be7e70463a5e37b703722d79c833d296ff/io/182064948/a16675f95a3f3c6b61307357be3c32be7e70463a5e37b703722d79c833d296ff-stdout with O_PATH: open /run/containerd/io.containerd.grpc.v1.cri/con
tainers/a16675f95a3f3c6b61307357be3c32be7e70463a5e37b703722d79c833d296ff/io/182064948/a16675f95a3f3c6b61307357be3c32be7e70463a5e37b703722d79c833d296ff-stdout: too many open files"
WARN[2021-04-21T10:07:55.435388176-04:00] Failed to Fetch k8s dns clusterIP exit status 1
The connection to the server localhost:8080 was refused - did you specify the right host or port?

WARN[2021-04-21T10:07:55.435428998-04:00] Using google dns 8.8.8.8
WARN[2021-04-21T10:07:55.616783855-04:00] Failed to Fetch k8s dns clusterIP exit status 1
The connection to the server localhost:8080 was refused - did you specify the right host or port?

WARN[2021-04-21T10:07:55.616812815-04:00] Using google dns 8.8.8.8
ERRO[2021-04-21T10:07:55.676765035-04:00] stock containerd failed to start UC           error="rpc error: code = Unknown desc = failed to create container io: failed to open /run/containerd/io.containerd.grpc.v1.cri/containers/4e09456ef058db9794e0ab37610a6359e004776c84c7559ac00229fa3b519b44/io/133637766/4e09456ef058db9794e0ab37610a6359e004776c84c7559ac00229fa3b519b44-stdout with O_PATH: open /run/containerd/io.containerd.grpc.v1.cri/containers/4e09456ef058db9794e0ab37610a6359e004776c84c7559ac00229fa3b519b44/io/133637766/4e09456ef058db9794e0ab37610a6359e004776c84c7559ac00229fa3b519b44-stdout: too many open files"
WARN[2021-04-21T10:07:55.877455054-04:00] Failed to Fetch k8s dns clusterIP exit status 1
The connection to the server localhost:8080 was refused - did you specify the right host or port?

WARN[2021-04-21T10:07:55.877491851-04:00] Using google dns 8.8.8.8
WARN[2021-04-21T10:07:55.962490807-04:00] Failed to Fetch k8s dns clusterIP exit status 1
The connection to the server localhost:8080 was refused - did you specify the right host or port?

WARN[2021-04-21T10:07:55.962533852-04:00] Using google dns 8.8.8.8
ERRO[2021-04-21T10:07:56.225404440-04:00] VM config for pod 2ecfef2261526a5afc22069926c7e928d9c2e9d55a1517f114d24e735cd6c127 does not exist
ERRO[2021-04-21T10:07:56.225433001-04:00]                                               error="VM config for pod does not exist"
ERRO[2021-04-21T10:07:56.234538488-04:00] VM config for pod 884e10c32b080de86f8f7b390f68e2e94faabe8a12d3ed52dfb51f199a67eb7d does not exist
ERRO[2021-04-21T10:07:56.234564359-04:00]                                               error="VM config for pod does not exist"
ERRO[2021-04-21T10:07:56.292045020-04:00] stock containerd failed to start UC           error="rpc error: code = Unknown desc = failed to create container io: failed to open /run/containerd/io.containerd.grpc.v1.cri/containers/d7276f89220d2d8bd91309606e7e826d3a87c431b041fa71b1514a08cf8dde0c/io/005718738/d7276f89220d2d8bd91309606e7e826d3a87c431b041fa71b1514a08cf8dde0c-stderr with O_PATH: open /run/containerd/io.containerd.grpc.v1.cri/containers/d7276f89220d2d8bd91309606e7e826d3a87c431b041fa71b1514a08cf8dde0c/io/005718738/d7276f89220d2d8bd91309606e7e826d3a87c431b041fa71b1514a08cf8dde0c-stderr: too many open files"
WARN[2021-04-21T10:07:56.429950650-04:00] Failed to Fetch k8s dns clusterIP exit status 1
The connection to the server localhost:8080 was refused - did you specify the right host or port?

WARN[2021-04-21T10:07:56.429995899-04:00] Using google dns 8.8.8.8
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x112a116]

goroutine 8132 [running]:
github.com/ease-lab/vhive/cri.(*Service).createUserContainer(0xc000480360, 0x14077a0, 0xc000fb9cb0, 0xc000fb9ce0, 0xc000f6ab10, 0x2, 0x2)
        /users/ahoste2/vhive/cri/container_create.go:94 +0x3d6
github.com/ease-lab/vhive/cri.(*Service).CreateContainer(0xc000480360, 0x14077a0, 0xc000fb9cb0, 0xc000fb9ce0, 0xc000480360, 0xc000fb9cb0, 0xc000f6aba0)
        /users/ahoste2/vhive/cri/container_create.go:53 +0x279
k8s.io/cri-api/pkg/apis/runtime/v1alpha2._RuntimeService_CreateContainer_Handler(0x12d90c0, 0xc000480360, 0x14077a0, 0xc000fb9cb0, 0xc0016b2cc0, 0x0, 0x14077a0, 0xc000fb9cb0, 0xc000f31600, 0xd72)
        /users/ahoste2/go/pkg/mod/k8s.io/cri-api@v0.16.16-rc.0/pkg/apis/runtime/v1alpha2/api.pb.go:7699 +0x214
google.golang.org/grpc.(*Server).processUnaryRPC(0xc0004ae000, 0x140f7e0, 0xc00049fc80, 0xc000e20100, 0xc000482660, 0x1abaaf0, 0x0, 0x0, 0x0)
        /users/ahoste2/go/pkg/mod/google.golang.org/grpc@v1.33.1/server.go:1210 +0x522
google.golang.org/grpc.(*Server).handleStream(0xc0004ae000, 0x140f7e0, 0xc00049fc80, 0xc000e20100, 0x0)
        /users/ahoste2/go/pkg/mod/google.golang.org/grpc@v1.33.1/server.go:1533 +0xd05
google.golang.org/grpc.(*Server).serveStreams.func1.2(0xc00019e070, 0xc0004ae000, 0x140f7e0, 0xc00049fc80, 0xc000e20100)
        /users/ahoste2/go/pkg/mod/google.golang.org/grpc@v1.33.1/server.go:871 +0xa5
created by google.golang.org/grpc.(*Server).serveStreams.func1
        /users/ahoste2/go/pkg/mod/google.golang.org/grpc@v1.33.1/server.go:869 +0x1fd
```

--- a/cri/container_create.go
+++ b/cri/container_create.go
@@ -91,6 +91,12 @@ func (s *Service) createUserContainer(ctx context.Context, r *criapi.CreateConta
 	// Wait for placeholder UC to be created
 	<-stockDone
 
+	// Check for error from container creation
+	if stockErr != nil {
+		log.WithError(stockErr).Error("failed to create container")
+		return nil, stockErr
+	}
+	
 	containerdID := stockResp.ContainerId
 	err = s.coordinator.insertActive(containerdID, funcInst)
 	if err != nil {

